membership.md

```diff
@@ -0,0 +1,170 @@
+- Expected to be responsive to review requests
+- Assigned PRs to review related to subproject of expertise
+- Assigned test bugs related to subproject of expertise
+- May get a badge on PR and issue comments
+
+## Approver
+
+Code approvers are able to both review and approve code contributions. While
+code review is focused on code quality and correctness, approval is focused on
+holistic acceptance of a contribution including: backwards / forwards
+compatibility, adhering to API and flag conventions, subtle performance and
+correctness issues, interactions with other parts of the system, etc.
+
+**Defined by:** *approvers* entry in an OWNERS file in a repo owned by the Kubeflow project.
+
+Approver status is scoped to a part of the codebase.
+
+### Requirements
+
+The following apply to the part of codebase for which one would be an approver in an [OWNERS] file (for repos using the bot).
+
+- Reviewer of the codebase for at least 3 months
+- Primary reviewer for at least 10 substantial PRs to the codebase
+- Reviewed or merged at least 30 PRs to the codebase
+- Nominated by a subproject owner
+  - With no objections from other subproject owners
+  - Done through PR to update the top-level OWNERS file
+
+**Note:** Working Group Leads may nominate and approve `Approvers` that don't meet these requirements due to exceptional circumstances. While acceptable in the short term, Working Group Leads should ensure that these `Approvers` eventually meet the requirements
+
+### Responsibilities and privileges
+
+The following apply to the part of codebase for which one would be an approver
+in an [OWNERS] file (for repos using the bot).
+
+- Approver status may be a precondition to accepting large code contributions
+- Demonstrate sound technical judgement
+- Responsible for project quality control via code reviews
+  - Focus on holistic acceptance of contribution such as dependencies with other features, backwards / forwards
+    compatibility, API and flag definitions, etc
+- Expected to be responsive to review requests
+- Mentor contributors and reviewers
+- May approve code contributions for acceptance
+
+## Inactive members
+
+*Members are continuously active contributors in the community.*
+
+A core principle in maintaining a healthy community is encouraging active
+participation. It is inevitable that people's focuses will change over time and
+they are not expected to be actively contributing forever.
+
+However, being a member of one of the Kubeflow GitHub organizations comes with
+an [elevated set of permissions]. These capabilities should not be used by those
+that are not familiar with the current state of the Kubeflow project.
+
+Therefore members with an extended period away from the project with no activity
+will be removed from the Kubeflow GitHub Organizations and will be required to
+go through the org membership process again after re-familiarizing themselves
+with the current state.
```

Comment:
I think, before removing inactive members from Kubeflow GitHub org, we should remove them from the OWNER files

Reply:
How about we formalize the inactivity time after which an approver/reviewer should be removed? If this is supposed to happen after X days, then we can automate this with bots eventually. Also, until we can implemented an automated system, who will periodically verify and clean up OWNERS files?

Reply:
I think 1-2 years without contributions is a starting point

Reply:
I updated it a bit (I've defined the period as 1 year)